### ISSUE TITLE
[Needs payment][Track Tax] Add new input fields for tax tracking in Money requests

### ISSUE THREAD

**Issue by MonilBhavsar** [Expensify/App#31672]
## Problem
Internal master Issue:https://github.com/Expensify/Expensify/issues/338212
Coming from this section in design doc https://docs.google.com/document/d/1nkusVPiYN3uYCJAH_2XknBv_oPpNOifVxheuxRvFTxs/edit?pli=1#bookmark=id.l319wrhremwa

This issue is divided in 2 phases to keep PR's small, test use cases efficiently and deliver in milestones

### Phase 1

In phase 1, we create a simple input fields that can allow inserting, editing and viewing tax details
As discussed in design doc, we need to create tax selector and tax amount components. Also add handles to update transaction tax details in Onyx and also pass tax details to API

Components and handlers should be created in a new way following this refactor PR https://github.com/Expensify/App/pull/28618

### Phase 2

In phase 2, we'll add more logic to tax input fields.
- Handling and displaying system messages when tax rates are updated
- When updating category, we need to apply category expense rule and hence update tax rate field and tax amount field accordingly
- As mentioned in design doc, only specific users can edit the tax details depending on the state of report

## Why is this important?

To allow users to track tax on the expenses, and migration users to newdot.

## Solution

Implement the section of the design doc

<details><summary>Issue Owner</summary>Current Issue Owner: @kadiealexander</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01fe3f2f6f3d94b969</li>
        <li>Upwork Job ID: 1785541230062321664</li>
        <li>Last Price Increase: 2024-05-01</li>
    </ul>
</details>

**Comment by MonilBhavsar** [comment]
Discussing here https://expensify.slack.com/archives/C03UK30EA1Z/p1700633473392829

**Comment by MonilBhavsar** [comment]
We're discussing next steps in slack. 
Artem Makushow and Eto Olei are going through design doc and will start working on it soon

**Comment by kadiealexander** [comment]
Not overdue!

**Comment by MonilBhavsar** [comment]
Draft PR is up!

**Comment by MonilBhavsar** [comment]
@teneeto this is the formulae to calculate tax which will be required.
```JS
const percentage = // selected by user from policyTaxRates_collection
const divisor = (percentage.slice(0, -1) / 100) + 1; // slice to remove % at the end; converts "10%" to "10"
return (parseInt(Math.round(amount - (amount / divisor)), 10)) / 10; // amount is the expense amount of transaction
```

**Comment by MonilBhavsar** [comment]
Assigning @alitoshmatov to this issue as they helped reviewing the PR

**Comment by MonilBhavsar** [comment]
We have merged first PR 🎉 
We'll now be working on second PR to view and allow updating fields

**Comment by MonilBhavsar** [comment]
Teeneto and I had a sync yesterday and teeneto will be working on second part of the issue

**Comment by MonilBhavsar** [comment]
Draft PR for Edit request is up!

**Comment by MonilBhavsar** [comment]
@teneeto the new API commands to call to update tax rate and tax amount are `UpdateMoneyRequestTaxRate` and `UpdateMoneyRequestTaxAmount`. Command takes common args such as transactionID and optimisticReportActionID and specific args - `taxCode: string` and `taxAmount: int` respectively 

Please refer to `UpdateMoneyRequestDate` command
We have to do in a similar way
https://github.com/Expensify/App/blob/70f3b99c0b5b53d8be84a07f6b0277cbfeda6df4/src/libs/actions/IOU.js#L1044

**Comment by alitoshmatov** [comment]
Automation created issue for a payment here - https://github.com/Expensify/App/issues/33441. 

@MonilBhavsar Should the payment be handled on that issue? I am just confused since this issue also exists

**Comment by MonilBhavsar** [comment]
We'll handle the payment here. Let's close that issue

**Comment by alitoshmatov** [comment]
Is draft PR ready for review or are we waiting for something else?

**Comment by teneeto** [comment]
> Is draft PR ready for review or are we waiting for something else?

Hi @alitoshmatov a draft PR is ready here: https://github.com/Expensify/App/pull/33927

Just waiting for a few BE changes to be finalised and then we'll continue AFFAIK.

**Comment by MonilBhavsar** [comment]
Yes correct!
We had to change backend _majorly_ and had to discuss. Discussion has concluded and it is being implemented now

**Comment by MonilBhavsar** [comment]
One backend PR is merged. We're waiting on other two to be merged

**Comment by MonilBhavsar** [comment]
We're still awaiting Web-E PR to be merged and deployed

**Comment by teneeto** [comment]
This also currently going through a small refactor to use `taxRates` from `policy_` rather than `policyTaxRates_` keys.

**Comment by MonilBhavsar** [comment]
Same^

**Comment by MonilBhavsar** [comment]
@alitoshmatov the second PR is ready for review! 
Thanks @teneeto 🙇

**Comment by alitoshmatov** [comment]
Is there any plan for our next step on this feature?

**Comment by MonilBhavsar** [comment]
We're working on other parallel issues. We can issue payment here and close this now

**Comment by alitoshmatov** [comment]
@kadiealexander Can we complete payment and close this one

**Comment by kadiealexander** [comment]
@MonilBhavsar I'm finding this hard to follow, could you please quickly summarise who is due payment for what?

**Comment by MonilBhavsar** [comment]
Payment summary: Pay $1000 to @alitoshmatov for reviewing two PR's created before the price update

1. https://github.com/Expensify/App/pull/33927
2. https://github.com/Expensify/App/pull/32550

There were some regressions coming out of it, but since it was a new feature and lots of new introductions to App, i think it's fine to pay full payment.

**Comment by MonilBhavsar** [comment]
Eto Olei is from Callstack and does not need payment

**Comment by kadiealexander** [comment]
@alitoshmatov I've sent you an upwork contract.

**Comment by kadiealexander** [comment]
Payouts due:

- [x] Contributor: $1000 @alitoshmatov

Upwork job is [here](upwork.com/jobs/~01fe3f2f6f3d94b969).
